EKF/ekf.h

```diff
@@ -57,58 +57,65 @@ class Ekf : public EstimatorInterface
+	void getGpsVelPosInnovVar(float hvel[2], float &vvel, float hpos[2], float &vpos);
 
-	// gets the innovations of the earth magnetic field measurements
-	void get_mag_innov(float mag_innov[3]);
+	void getGpsVelPosInnovRatio(float& hvel, float &vvel, float& hpos, float &vpos);
 
-	// gets the innovations of the heading measurement
-	void get_heading_innov(float *heading_innov);
+	void getEvVelPosInnov(float hvel[2], float& vvel, float hpos[2], float& vpos);
 
-	// gets the innovation variances of velocity and position measurements
-	// 0-2 vel, 3-5 pos
-	void get_vel_pos_innov_var(float vel_pos_innov_var[6]);
+	void getEvVelPosInnovVar(float hvel[2], float &vvel, float hpos[2], float &vpos);
 
-	// gets the innovation variances of the earth magnetic field measurements
-	void get_mag_innov_var(float mag_innov_var[3]);
+	void getEvVelPosInnovRatio(float& hvel, float &vvel, float& hpos, float &vpos);
 
-	// gets the innovations of airspeed measurement
-	void get_airspeed_innov(float *airspeed_innov);
+	void getAuxVelInnov(float aux_vel_innov[2]);
 
-	// gets the innovation variance of the airspeed measurement
-	void get_airspeed_innov_var(float *airspeed_innov_var);
+	void getAuxVelInnovVar(float aux_vel_innov[2]);
 
-	// gets the innovations of synthetic sideslip measurement
-	void get_beta_innov(float *beta_innov);
+	void getAuxVelInnovRatio(float &aux_vel_innov_ratio);
 
-	// gets the innovation variance of the synthetic sideslip measurement
-	void get_beta_innov_var(float *beta_innov_var);
+	void getFlowInnov(float flow_innov[2]);
 
-	// gets the innovation variance of the heading measurement
-	void get_heading_innov_var(float *heading_innov_var);
+	void getFlowInnovVar(float flow_innov_var[2]);
 
-	// gets the innovation variance of the flow measurement
-	void get_flow_innov_var(float flow_innov_var[2]);
+	void getFlowInnovRatio(float &flow_innov_ratio);
 
-	// gets the innovation of the flow measurement
-	void get_flow_innov(float flow_innov[2]);
+	void getHeadingInnov(float &heading_innov);
 
-	// gets the innovation variance of the drag specific force measurement
-	void get_drag_innov_var(float drag_innov_var[2]);
+	void getHeadingInnovVar(float &heading_innov_var);
 
-	// gets the innovation of the drag specific force measurement
-	void get_drag_innov(float drag_innov[2]);
+	void getHeadingInnovRatio(float &heading_innov_ratio);
 
-	// gets the innovation variance of the HAGL measurement
-	void get_hagl_innov_var(float *hagl_innov_var);
+	void getMagInnov(float mag_innov[3]);
 
-	// gets the innovation of the HAGL measurement
-	void get_hagl_innov(float *hagl_innov);
+	void getMagInnovVar(float mag_innov_var[3]);
+
+	void getMagInnovRatio(float &mag_innov_ratio);
+
+	void getDragInnov(float drag_innov[2]);
+
+	void getDragInnovVar(float drag_innov_var[2]);
+
+	void getDragInnovRatio(float drag_innov_ratio[2]);
+
+	void getAirspeedInnov(float &airspeed_innov);
+
+	void getAirspeedInnovVar(float &airspeed_innov_var);
```

Comment:
If you're going to bother changing these how about we return by value whenever possible? A lot of them could then turn into inlined getters.

Reply:
That is a nice idea. A lot of the float arrays that are returned are directly put into message fields, which are also a float array. So returning them by value as a Vector would be a lot of effort. 
There are a lot of other functions that do not follow the requested standard and I suggest that we try to clean this up in a separate PR.